packages/navie/src/services/message-token-reducer-service/token-count-strategies/claude-token-count-error-parser.ts

```diff
@@ -0,0 +1,14 @@
+import Message from '../../../message';
+import { ApiError, TokenCountStrategy } from '../interfaces';
+
+export default class ClaudeTokenCountErrorParser implements TokenCountStrategy {
+  countTokens(_messages: Message[], _model: string, apiError?: ApiError): number | undefined {
+    if (!apiError) return undefined;
+
+    const match = /prompt is too long: (\d+) tokens > \d+ maximum/g.exec(apiError.message);
+    if (!match) return undefined;
+
+    const tokenLimit = parseInt(match[1], 10);
+    return tokenLimit;
```

Comment:
This will return `NaN` if it can't parse (which won't be caught by `!== undefined` above). Consider changing the test there to truthiness test (`NaN` is falsy), or check here:
```suggestion
    return parseInt(match[1], 10) || undefined;
```

(There are a couple of other instances of the same problem.)